File: create_placeholders.js
const fs = require('fs');
const path = require('path');

// 确保images目录存在
const imagesDir = path.join(__dirname, 'images');
if (!fs.existsSync(imagesDir)) {
    fs.mkdirSync(imagesDir);
}

// 简单的SVG图片数据（转换为base64）
const imagesData = [
    {
        filename: 'habitat-map.jpg',
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="800" height="500" viewBox="0 0 800 500"><rect width="800" height="500" fill="#4a7c59"/><text x="400" y="250" font-family="Arial, sans-serif" font-size="30" fill="white" text-anchor="middle">Habitat Map</text></svg>`
    },
    {
        filename: 'endangered-species.jpg', 
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="600" height="400" viewBox="0 0 600 400"><rect width="600" height="400" fill="#e74c3c"/><text x="300" y="200" font-family="Arial, sans-serif" font-size="30" fill="white" text-anchor="middle">Endangered Species</text></svg>`
    },
    {
        filename: 'deforestation.jpg',
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="400" height="300" viewBox="0 0 400 300"><rect width="400" height="300" fill="#f39c12"/><text x="200" y="150" font-family="Arial, sans-serif" font-size="24" fill="white" text-anchor="middle">Deforestation</text></svg>`
    },
    {
        filename: 'habitat-fragmentation.jpg',
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="400" height="300" viewBox="0 0 400 300"><rect width="400" height="300" fill="#e67e22"/><text x="200" y="150" font-family="Arial, sans-serif" font-size="24" fill="white" text-anchor="middle">Habitat Fragmentation</text></svg>`
    },
    {
        filename: 'poaching.jpg',
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="400" height="300" viewBox="0 0 400 300"><rect width="400" height="300" fill="#c0392b"/><text x="200" y="150" font-family="Arial, sans-serif" font-size="24" fill="white" text-anchor="middle">Poaching</text></svg>`
    },
    {
        filename: 'pollution.jpg',
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="400" height="300" viewBox="0 0 400 300"><rect width="400" height="300" fill="#8e44ad"/><text x="200" y="150" font-family="Arial, sans-serif" font-size="24" fill="white" text-anchor="middle">Pollution</text></svg>`
    },
    {
        filename: 'conservation-action.jpg',
        svg: `<svg xmlns="http://www.w3.org/2000/svg" width="600" height="400" viewBox="0 0 600 400"><rect width="600" height="400" fill="#27ae60"/><text x="300" y="200" font-family="Arial, sans-serif" font-size="30" fill="white" text-anchor="middle">Conservation Action</text></svg>`
    }
];

// 将SVG转换为Buffer并写入文件
imagesData.forEach(data => {
    const filePath = path.join(imagesDir, data.filename);
    // 将SVG转换为Buffer
    const buffer = Buffer.from(data.svg);
    fs.writeFileSync(filePath, buffer);
    console.log(`Created ${data.filename}`);
});

console.log('All placeholder images have been created successfully!');

File: create_placeholders.py
import os
from PIL import Image, ImageDraw, ImageFont

# 创建图片目录
os.makedirs('images', exist_ok=True)

# 定义图片配置
images_config = [
    {'filename': 'habitat-map.jpg', 'size': (800, 500), 'color': (74, 124, 89), 'text': 'Habitat Map'},
    {'filename': 'endangered-species.jpg', 'size': (600, 400), 'color': (231, 76, 60), 'text': 'Endangered Species'},
    {'filename': 'deforestation.jpg', 'size': (400, 300), 'color': (243, 156, 18), 'text': 'Deforestation'},
    {'filename': 'habitat-fragmentation.jpg', 'size': (400, 300), 'color': (230, 126, 34), 'text': 'Habitat Fragmentation'},
    {'filename': 'poaching.jpg', 'size': (400, 300), 'color': (192, 57, 43), 'text': 'Poaching'},
    {'filename': 'pollution.jpg', 'size': (400, 300), 'color': (142, 68, 173), 'text': 'Pollution'},
    {'filename': 'conservation-action.jpg', 'size': (600, 400), 'color': (39, 174, 96), 'text': 'Conservation Action'}
]

def create_placeholder_image(filename, size, bg_color, text):
    # 创建带有指定背景色的图像
    img = Image.new('RGB', size, bg_color)
    draw = ImageDraw.Draw(img)
    
    # 尝试使用默认字体，如果不可用则使用内置字体
    try:
        # 在Windows系统上常用的字体
        font = ImageFont.truetype("arial.ttf", 40)
    except:
        try:
            # 尝试其他常见字体
            font = ImageFont.truetype("C:/Windows/Fonts/arial.ttf", 40)
        except:
            # 使用默认字体
            font = ImageFont.load_default()
    
    # 获取文本尺寸
    text_bbox = draw.textbbox((0, 0), text, font=font)
    text_width = text_bbox[2] - text_bbox[0]
    text_height = text_bbox[3] - text_bbox[1]
    
    # 计算文本位置（居中）
    x = (size[0] - text_width) / 2
    y = (size[1] - text_height) / 2
    
    # 绘制文本
    draw.text((x, y), text, fill=(255, 255, 255), font=font)
    
    # 保存图像
    img.save(f'images/{filename}')
    print(f'Created {filename}')

# 创建所有占位图片
for config in images_config:
    create_placeholder_image(**config)

print('All placeholder images have been created successfully!')